src/Xamarin.Android.Build.Tasks/Utilities/ManifestDocument.cs

```diff
@@ -282,7 +281,7 @@ public IList<string> Merge (TaskLoggingHelper log, TypeDefinitionCache cache, Li
 			if (tsv != null)
 				targetSdkVersion = tsv.Value;
 			else {
-				targetSdkVersion = SdkVersionName;
+				uses.SetAttributeValue (androidNs + "targetSdkVersion", targetSdkVersion = SdkVersionName);
```

Comment:
@jonpryor  this is the main behavioral change here ^^

Before these changes, the absence of `<uses-sdk/>` worked differently than if you had an empty `<uses-sdk/>`.

Reply:
@jonathanpeppers wrote in the PR message:

> Reviewing the code in ManifestDocument, it seems like we should be
> replacing the targetSdkVersion for empty <uses-sdk/> elements?

Notably absent is *why* we "should" do that.

See also: https://github.com/xamarin/monodroid/commit/1c78098c5505a3e15ef2e59a9ca8f3e53b6da431

>     eno:  @jonp what behavior does Android actually change?
>     jonp: what doesn't it change? :-)
>     jonp: @eno seems that the "Android X APIs" page also lists semantic
>           changes that depend on minSdkVersion, e.g.:
>           http://developer.android.com/about/versions/android-4.2.html
>           "Content providers... This change takes effect only if you set
>           either android:targetSdkVersion or android:minSdkVersion to 17
>           or higher."
> 
> In short, //uses-sdk/@android:minSdkVersion controls EVERYTHING.

`targetSdkVersion` is in a similar position: *it's presence* (or lack thereof!) and value will change the semantics of Android behavior.

For a more recent example consider [Android 6.0's introduction of Runtime Permissions](https://developer.android.com/about/versions/marshmallow/android-6.0-changes#behavior-runtime-permissions).  The new behavior was *only* enabled when `targetSdkVersion` was >= 23, but for apps which had to update this value, it was a potentially significant amount of work to "fix" the app after that change.

In short, the *lack* of `//uses-sdk/@android:minSdkVersion` and/or `//uses-sdk/@android:targetSdkVersion` is *semantically meaningful*.  *At minimum*, there needs to be *a* way to not have these attributes, if only for our own testing ("what happens to app behavior of we don't set these properties?").

"Differently" concerning to me is the underlying scenario: reference the `Xamarin.AndroidX.Media`, and your apps `//uses-sdk` attribute values are... *what*? Replaced by the values from `Xamarin.AndroidX.Media`?  Used if present in `Properties\AndroidManifest.xml`, but otherwise come from "somewhere", including `Xamarin.AndroidX.Media`?  Guess this means that the [manifestmerger docs](https://developer.android.com/studio/build/manifest-merge) are now effectively *required reading* -- and has been since 2019? -- which isn't something that I had realized.  Which is "differently joyful"?

> * Attributes in the `<uses-sdk>` element always use the value from the higher-priority manifest, except in the following situations:

which seems to mean that `//uses-sdk/@android:targetSdkVersion` & co. are no longer *fully* controllable by the developer, meaning I *can't* write a one-off "personal" app that ignores the Runtime Permissions checks and *also* uses AndroidX, as the inclusion of some "random" AndroidX lib will *force* a `targetSdkVersion` value >= 23.

…which might be "fine", I guess?  (I'm not necessarily *thrilled*, but this might be "fine"?)

Reply:
That analysis out of the way, what should we *do* about it?

The ship has sailed on manifestmerger; it's been in since d16-5, so apparently this change hasn't "broken" anybody, and this is presumably more "in spirit" with what Java requires.

If `Properties\AndroidManifest.xml` *does* contain a `//uses-sdk/@android:targetSdkVersion` value, *and* it's being "ignored" (because manifestmerger is updating it based on dependencies), we should *still* emit an XA1006 warning.  (This might be the case in this PR, but I'm not sure.  This should be verified.)  Why?  Because the `Properties\AndroidManifest.xml` value is the "source of truth" as provided by the developer, and if this source of truth is being ignored -- which it is -- then they should be told so.  The warning is useful, *and* can be disabled by updating the value within `Properties\AnsroidManifest.xml`.

This new behavior also needs to be adequately documented & described, particularly the relation to manifestmerger (which wasn't explicitly obvious).

Reply:
@jonpryor so the only issue here is an app has an empty `<uses-sdk/>` in their `AndroidManifest.xml`. I would guess that almost no one does this. The templates have defined `minSdkVersion` and `targetSdkVersion` for a long time--maybe always?

However, our MSBuild tests use `<uses-sdk/>` -- which is kind of odd. I think the fix might should just change these tests to use:

```xml
<uses-sdk android:minSdkVersion="{minSupportedApi}" android:targetSdkVersion="{latestStableApi}" />
```

And I won't change any behavior here.